### ISSUE TITLE
Update API and/or documentation for cognitoIdentity.setIdentityPoolRoles()

### ISSUE THREAD

**Issue by matsev** [aws/aws-sdk-js#905]
In the documentation of [CognitoIdentity.setIdentityPoolRoles(params = {}, callback)](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CognitoIdentity.html#setIdentityPoolRoles-property) it is stated that:

> params (Object) (defaults to: {}) —
IdentityPoolId — (String)
An identity pool ID in the format REGION:GUID.

This information is wrong, any attempt will cause an unauthenticated exception. 

Instead, by providing the _ARN_ to the identity pool (despite that the key is called `IdentityPoolId`) I managed to change the `authenticated` and `unauthenticated` roles as expected. 

May I also suggest that you clarify that it is possible to remove the roles from an identity pool by passing an empty roles document, e.g. `"Roles" : {}`.

Version: `var cognitoIdentity = new AWS.CognitoIdentity({apiVersion: '2014-06-30'});`

**Comment by chrisradek** [comment]
@matsev,
Does the `IdentityPoolId` you ended up using match the pattern `REGION:GUID`?
If you look up the Identity Pool Id in the console, you'll see it in the pattern `REGIONGUID`, with no colon between the two. The ARN would actually look very different: `arn:aws:cognito-identity:region:accountId:identitypool/REGIONGUID`.

If you call `listIdentityPools({MaxResults:60});`, you are returned the `IdentityPoolId`s in the `REGION:GUID` format that the API calls require.

We can add a note to specify that calling this operation will set the roles to exactly what is provided, and doesn't simply update them.

**Comment by cmawhorter** [comment]
I think I'm seeing some key name weirdness too.  

I'm using aws-sdk 2.2.37 in browser and server (mocha tests) and there seems to be some inconsistency with behavior between browser and web.

The [documentation](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CognitoIdentityCredentials.html) says the identity pool id key is `IdentityPoolId`.  This does not work in node.

tl;dr; use both `IdentityPoolId` and `IdentityId` when constructing your cognito credentials object.

In browser:

```js
var creds = new AWS.CognitoIdentityCredentials({
    IdentityPoolId: "us-east-1:78fc........75",
    RoleArn: 'arn:aws:iam::17xxxxxxx9:role/Cognito_identitypoolnameAuth_Role', 
    Logins: logins
  })
```

This is working and creates a request to cognito with a body that is `IdentityPool` and NOT `IdentityPoolId`.  This same key translation does not seem to occur in node.

![image](https://cloud.githubusercontent.com/assets/142338/13195554/42770310-d76a-11e5-8234-078831f5c6b5.png)

However, using that same config in node it fails I get a `IdentityPool 'us-east-1:78fc.........75' not found.` error.

Changing `IdentityPoolId` to `IdentityId` in node causes the request to succeed.

http 400 w/`IdentityPoolId`:
![image](https://cloud.githubusercontent.com/assets/142338/13195584/1267ab06-d76b-11e5-9c0f-bb04cf054604.png)

http 200 w/`IdentityId`:
![image](https://cloud.githubusercontent.com/assets/142338/13195594/30789f2e-d76b-11e5-8cc2-0f82a61f0960.png)

**Comment by cmawhorter** [comment]
i take some of that back... it varies from identity pool to identity pool?  yikes.

**Comment by matsev** [comment]
@chrisradek @cmawhorter This morning, I revisited the issue that I created last week and I am unable to reproduce the error as I reported it. I am sorry for spending your time.

**Comment by cmawhorter** [comment]
i'm seeing a similar issue to you.  i'll open a separate issue
